### ISSUE TITLE
geo URIs: separate z parameter with a semicolon, not question mark

### ISSUE THREAD

**Issue by eserte** [openstreetmap/openstreetmap-website#5322]
### How to reproduce the issue?

It seems that additional parameters in a geo URI should be separated by semicolons, not by a question mark.
Currently the share functionality on the openstreetmap website is producing geo URIs like this:
```
geo:52.51626,13.37787?z=15
```
However, all documentation about geo URIs I found (https://en.wikipedia.org/wiki/Geo_URI_scheme, https://www.rfc-editor.org/rfc/rfc5870.txt) use only semicolons. Also
I found a software (a geolink extension for emacs org-mode) which also thinks it should be semicolons:
https://github.com/misohena/org-geolink/blob/main/org-geolink.el#L31-L33

So should the share functionality generate the above as
```
geo:52.51626,13.37787;z=15
```
?

**Comment by tomhughes** [comment]
Aren't `?` and `;` both allowed in URL syntax to introduce a parameter string? Though `?` is pretty much always what people use to the extent that I can't really ever remember anybody using `;` instead!

**Comment by tomhughes** [comment]
So [RFC 5870](https://datatracker.ietf.org/doc/html/rfc5870) does use semicolon to introduce geo URI parameters but `z` is not a defined parameter, it is extra optional information that we are passing using the [generic query syntax from RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.4).

**Comment by eserte** [comment]
> Aren't ? and ; both allowed in URL syntax to introduce a parameter string? Though ? is pretty much always what people use to the extent that I can't really ever remember anybody using ; instead!

In traditional `http` URIs, only `?` is used to start the parameters, and `;` or `&` are used to separate key-value pairs.

But `geo` URIs are not `http` URIs.

**Comment by tomhughes** [comment]
Yes I was misremembering what semicolon is used for in URLs earlier.

It's still the case I think that all URIs support query string parameters in principle, and that geo URIs have no standardised way of indicating zoom level so there's no real right answer but compatibility with what Google do is important given that was the main reason for adding them.
